File: 品味Java的21種設計模式/dp_cpp/abstractfactory/example1/MSIMainboard.cpp
#include "MSIMainboard.h"
#include <iostream>

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace abstractfactory
			{
				namespace example1
				{

					MSIMainboard::MSIMainboard(int cpuHoles)
					{
						this->cpuHoles = cpuHoles;
					}

					void MSIMainboard::installCPU()
					{
                        std::cout << "now in MSIMainboard,cpuHoles=" << cpuHoles;
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/abstractfactory/example1/MSIMainboard.java
package cn.javass.dp.abstractfactory.example1;
/**
 * 微星的主板
 */
public class MSIMainboard implements MainboardApi {
    /**
     * CPU插槽的孔数
     */
    private int cpuHoles = 0;
    /**
     * 构造方法，传入CPU插槽的孔数
     * @param cpuHoles CPU插槽的孔数
     */
    public MSIMainboard(int cpuHoles) {
        this.cpuHoles = cpuHoles;
    }
    public void installCPU() {
        System.out.println("now in MSIMainboard,cpuHoles=" + cpuHoles);
    }
}
